WordPress/gutenberg
# Theme.json Schema: add support for button elements

## What?

In https://github.com/WordPress/gutenberg/pull/40260, we added support for button elements to `theme.json`. We now need to update the `schema` for theme.json to reflect this accurately.

## Why?

This improves the developer experience by automatically validating the contents of `theme.json`.

## How?

By including `button` as one of the properties listed under the `elements` section.

I copied the way we're already doing this for `core/button` [here](https://github.com/WordPress/gutenberg/blob/trunk/schemas/json/theme.json#L1148), which has worked in my testing, but I'm not sure if there's a better/more correct way to handle this.

## Testing Instructions




1. Activate a theme which you can modify locally.
2. Open the theme's theme.json file and update the `$schema` to point to your local Gutenberg version (follow the instructions from the docs [here](https://github.com/WordPress/gutenberg/tree/trunk/schemas#local-development)).
3. Add some button rules under `elements.button` and check your IDE doesn't complain. Here is an example:
```
"elements": {
	"button": {
		"border": {
			"radius": "0",
			"width": "0"
		},
		"color": {
			"background": "black",
			"text": "white"
		},
		"typography": {
			"textTransform": "uppercase"
		},
		"spacing": {
			"padding": {
				"top": "10px",
				"bottom": "10px",
				"left": "20px",
				"right": "20px"
			}
		}
	}
}
```
4. Add some invalid button properties and check that it does complain (e.g. `textDecoration`).
